scoutcloud/scoutcloud/src/logic/config/variables/rpc_url.rs

```diff
@@ -0,0 +1,143 @@
+
+pub struct RpcUrl(Url);
+
+#[async_trait::async_trait]
+impl UserVariable<Url> for RpcUrl {
+    fn new(v: Url, _config: &ConfigValidationContext) -> Result<Self, Error> {
+        Ok(Self(v))
+    }
+
+    async fn build_config_vars(
+        &self,
+        _config: &ConfigValidationContext,
+    ) -> Result<Vec<ParsedVariable>, Error> {
+        let mut parsed = vec![];
+
+        // check json rpc
+        let provider = Provider::<Http>::try_from(self.0.as_str())
+            .context("failed to parse url as http")
+            .map_err(|e| Error::Validation(e.to_string()))?;
+
+        check_jsonrpc_health(&provider)
+            .await
+            .map_err(|e| Error::Validation(e.to_string()))?;
+        parsed.push((
+            ParsedVariableKey::BackendEnv("ETHEREUM_JSONRPC_HTTP_URL".to_string()),
+            serde_json::Value::String(self.0.to_string()),
+        ));
+
+        // check trace method
+        // TODO: check trace method according to node_type
+        match check_any_trace_method(&provider).await {
+            Ok(()) => {
+                parsed.push((
+                    ParsedVariableKey::BackendEnv("ETHEREUM_JSONRPC_TRACE_URL".to_string()),
+                    serde_json::Value::String(self.0.to_string()),
+                ));
+                parsed.push((
+                    ParsedVariableKey::BackendEnv(
+                        "INDEXER_DISABLE_INTERNAL_TRANSACTIONS_FETCHER".to_string(),
+                    ),
+                    serde_json::Value::String("false".to_string()),
+                ));
+            }
+            Err(err) => {
+                tracing::warn!(
+                    err =? err,
+                    "`rpc_url` does not support tracing, disabling internal transactions"
+                );
+                parsed.push((
+                    ParsedVariableKey::BackendEnv(
+                        "INDEXER_DISABLE_INTERNAL_TRANSACTIONS_FETCHER".to_string(),
+                    ),
+                    serde_json::Value::String("true".to_string()),
+                ));
+            }
+        };
+
+        // check websocket
+        if let Some(ws_url) = get_any_healthy_ws_url(self.0.clone())
+            .await
+            .map_err(Error::Internal)?
+        {
+            parsed.push((
+                ParsedVariableKey::BackendEnv("ETHEREUM_JSONRPC_WS_URL".to_string()),
+                serde_json::Value::String(ws_url.to_string()),
+            ));
+        } else {
+            tracing::warn!(
+                "no valid websocket url found for `rpc_url`, skipping websocket configuration"
+            );
+        }
+        Ok(parsed)
+    }
+}
+async fn check_jsonrpc_health(provider: &Provider<Http>) -> Result<(), anyhow::Error> {
+    provider
+        .get_chainid()
+        .await
+        .map_err(|e| anyhow::anyhow!("failed to check health of `rpc_url`: {e}"))?;
+
+    Ok(())
+}
+
+async fn check_any_trace_method(provider: &Provider<Http>) -> Result<(), anyhow::Error> {
```

Comment:
Why do we need to check all tracing methods and why isn't the info which tracing method have worked is not used further?

Reply:
this is temp solution. we check any trace method and if there is not found trace method, we disable internal txns indexer